mysqlnd_azure.c

```diff
@@ -634,12 +711,24 @@ MYSQLND_METHOD(mysqlnd_azure, connect)(MYSQLND * conn_handle,
             if (!(temp_flags & CLIENT_SSL)) {
                 //REDIRECT_ON, no ssl, return error
                 if((MYSQLND_AZURE_G(enableRedirect) == REDIRECT_ON)) {
+                  if (logLevel > 2) {
+                    AZURE_LOG("DEBUG", "mysqlnd_azure.enableRedirect: ON");
+                  }
+                  if (logLevel > 0) {
+                    AZURE_LOG("ERROR", "SSL not enabled when mysqlnd_azure.enableRedirect if ON");
+                  }
                     SET_CLIENT_ERROR((*pconn)->error_info, MYSQLND_AZURE_ENFORCE_REDIRECT_ERROR_NO, UNKNOWN_SQLSTATE, "mysqlnd_azure.enableRedirect is on, but SSL option is not set in connection string. Redirection is only possible with SSL.");
                     (*pconn)->m->local_tx_end(*pconn, this_func, FAIL);
                     (*pconn)->m->free_contents(*pconn);
                     return FAIL;
```

Comment:
Here is also a return. Need to   CLOSE_LOGFILE(); here too

Reply:
Your are right.

But I don't know why use `return FAIL;` here and `DBG_RETURN(ret);` at the end?

Reply:
Yeah, DEG_RETURN(FAIL) should be better